src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -72,6 +78,13 @@ public sealed class RestoreManagerPackage : AsyncPackage
                 _buildEvents = dte.Events.BuildEvents;
                 _buildEvents.OnBuildBegin += BuildEvents_OnBuildBegin;
 
+                var solutionBuildManager = (IVsSolutionBuildManager5)await GetServiceAsync(typeof(SVsSolutionBuildManager));
+                if (solutionBuildManager != null)
+                {
+                    uint updateSolutionEventsCookie4;
```

Comment:
Packages absolutely get disposed of on VS exit. Why do you think that doesn't happen?
There are occasionally memory leak runs that people check on VS exit, and advise calls without matching unadvise calls cause leaks to be reported. So even if you have it for the life of VS, it's good to balance your Advise calls with Unadvise calls.